pandas/core/groupby/grouper.py

```diff
@@ -491,8 +491,11 @@ def get_grouper(
                     raise ValueError("multiple levels only valid with MultiIndex")
 
             if isinstance(level, str):
-                if obj.index.name != level:
-                    raise ValueError(f"level name {level} is not the name of the index")
+                axis_name = obj._get_axis_name(axis)
```

Comment:
I think just use `obj._get_axis`.
```suggestion
                if obj._get_axis(axis) != level:
```

Reply:
leads to the following error:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

Reply:
should work with obj._get_axis(axis).name